### PR TITLE
Equipment details

### PR DESCRIPTION
**Pull request by nonifier**
This change aims at displaying correctly the `/equipment_reports` new endpoint. It shows:
- for each `equipment_report`: the line and number of stop areas with equipment.
- for each `stop_area_equipments`: the stop area with the number of equipment. Along with number of unavailable equipment in red.
- for each `equipment_details`: the name, status, cause & effect of the equipment detail.

image

### CODE REVIEW

File: js/summary.js [hunk 1]
```diff
@@ -658,6 +658,55 @@ summary.make.line_report = function(context, json) {
...
+    var res = $('<span>')
+        .append('stop_area: ')
+        .append(summary.run(context, 'stop_area', json.stop_area))
+        .append(sprintf(', %d equipments', equip_details.length));
+
+    var unavailable_equipments = equip_details.filter(function(o){
+        return o.current_availability.status === 'unavailable';
+    }).length;
+    if(unavailable_equipments) {
+        res.append($('<span/>').css('color', 'red').text(' ('+unavailable_equipments+' unavailable)'));
```
Comment (TeXitoi) [inline js/summary.js]: please no hard coded colors: https://github.com/CanalTP/navitia-playground/blob/master/js/summary.js#L361

File: js/extended.js [hunk 2]
```diff
@@ -308,6 +308,23 @@ extended.make.vehicle_journey = function(context, json) {
     return res;
 };
 
+extended.make.equipment_report = function(context, json) {
+    var res = $('<div class="list"/>');
```
Comment (TeXitoi) [inline js/extended.js]: Why `class="list"`?

File: js/summary.js [hunk 3]
```diff
@@ -658,6 +658,55 @@ summary.make.line_report = function(context, json) {
...
+        .append(json.name || json.id).append(', ')
+        .append('type: '+ json.embedded_type);
+
+    var cur_avail = json.current_availability;
+    if (cur_avail) {
+        if(cur_avail.status) {
+            var colours = {
+                'unavailable' : 'red',
+            };
+            var colour = (colours[cur_avail.status] || 'grey');
```
Comment (TeXitoi) [inline js/summary.js]: Better to use style for that.

File: js/summary.js (hunk 1 above)
Comment (nonifier) [reply js/summary.js]: that was ugly indeed.

File: js/summary.js (hunk 3 above)
Comment (nonifier) [reply js/summary.js]: :+1:

**Comment by nonifier** [comment]
@pbougue : Regarding the status, having it first would be really weird as most of the equipments come from the data and will have an `unknown` status. As opposed to disruptions that have a name associated wit the severity.
I can put name, cause, effet on a comma-separated one-liner. Although, naming is crap for the Sytral, they have very looooooooooooooooooong names.
I agree that I would be nicer to have the details in an Ext section, I can add this.

**Pull request merged**